### ISSUE TITLE
'Anchor / querystring' drops value in Single URL Picker

### ISSUE THREAD

**Issue by dotbenstar** [umbraco/Umbraco-CMS#5601]
When using a Single URL Picker data type anything I put into 'Anchor / querystring' is discarded. 

This is using Umbraco 7.13.2

In a rich text editor data types this functionality works correctly.

![umbraco-url-picker-anchor](https://user-images.githubusercontent.com/20684924/59033047-a9969700-885f-11e9-8f15-c6a123617965.jpg)

This screen recording shows the bug in action: https://youtu.be/GdgL7gWskAE

1. Using a document type with a Single URL Picker data type, add a link (note: you can't just add a relative anchor link, you must use an external link or pick an internal link first - see issue [5295](https://github.com/umbraco/Umbraco-CMS/issues/5295))
2. Type in an Anchor value
3. Click Select
4. Edit the link again and the value is gone.

**Comment by nul800sebastiaan** [comment]
Thanks, I'll close this as a duplicate of #5295 

It's on our list to fix this so please follow issue #5295 for updates.

**Comment by dotbenstar** [comment]
I've just had version 7.15.2 deployed for testing and this issue is not fixed.

Using a Single URL Picker data type, you cannot link to an anchor.

#5295 fixes it so that when using the rich text editor, you can now link to anchors on the same page without having to fill out the internal link in full (traversing the site tree for the page you're already on.) That fix is lovely.

The issue i was raising here completely removes the ability to use anchor links in the Single URL Picker data types - it wasn't just about the workaround being time consuming.

10 seconds into the video demonstrates it. https://youtu.be/GdgL7gWskAE?t=10. I edit an existing link to add an anchor. Then go back into edit it and the anchor value is gone.

**Comment by rickbracedigital** [comment]
Hi,

Has this issue been fixed? I am seeing this on v 7.15.3.

Thanks,
Rick

**Comment by jayrdi** [comment]
Same here, version 7.15.5. I'm trying to link to a specific section on a page by using the "Anchor/querystring" field in Single URL Picker, as @dotbenstar raised above and, on saving, it removes the querystring.

Even ignoring the ID given to the section and selecting the specific Page Section node I'm trying to link to results in link that ends ".../#pos_59690", which just links to the top of the page (parent node).

Could this please be reopened? I'm not sure why it was closed...

Thanks,
John

**Comment by tomvanenckevort** [comment]
I came across this issue on a v7.14.0 site today, but in our case it was because the data type of our single URL picker on this site was using RJP MultiUrlPicker (so not the built-in Umbraco picker, as this site was orginially built before that was added in).

After upgrading RJP MultiUrlPicker from v2.1.0 to v2.2.1 the issue went away for us.

It might not apply to the folks who have reported this issue on v7.15.x sites if they're not using that package, but it might help someone who is seeing this issue on an older site.
